pkg/internal/credentials.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/kubevirt"
+
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Credentials contains the necessary kubevirt credential information.
+type Credentials struct {
+	CloudProviderConfig string
+}
+
+// GetCredentials computes for a given context and infrastructure the corresponding credentials object.
+func GetCredentials(ctx context.Context, c client.Client, secretRef corev1.SecretReference) (*Credentials, error) {
```

Comment:
Why do you need these credentials, and how do you intend to use them? Right now the `GetCredentials` method is not used. Where do you intend to use it?
Also please have in mind that the `cloudprovider` secret that is referenced in the infra resource doesn't have a `cloud-provider-config` key at all. The only key it has is `kubeconfig` and it contains the kubeconfig of the "provider" cluster (where VMs, etc. are created).